controllers/apiserver.go

```diff
@@ -39,6 +45,108 @@ var samplePipelineTemplates = map[string]string{
+
+	item := make(map[string]string)
+
+	// Get Required Fields
+	pName, err := config.GetStringConfigOrDie(fmt.Sprintf("ManagedPipelinesMetadata.%s.Name", pipeline))
+	if err != nil {
+		return nil, err
+	}
+	pFile, err := config.GetStringConfigOrDie(fmt.Sprintf("ManagedPipelinesMetadata.%s.Filepath", pipeline))
+	if err != nil {
+		return nil, err
+	}
+
+	// Get optional fields
+	pDesc := config.GetStringConfigWithDefault(fmt.Sprintf("ManagedPipelinesMetadata.%s.Description", pipeline), "")
+	pVerName := config.GetStringConfigWithDefault(fmt.Sprintf("ManagedPipelinesMetadata.%s.VersionName", pipeline), "")
+	pVerDesc := config.GetStringConfigWithDefault(fmt.Sprintf("ManagedPipelinesMetadata.%s.VersionDescription", pipeline), "")
+
+	// Create Sample Config item
+	item["name"] = pName
+	item["file"] = pFile
+	item["description"] = pDesc
+	item["versionName"] = pVerName
+	item["versionDescription"] = pVerDesc
+
+	return item, nil
+
+}
+
+func (r *DSPAReconciler) GetSampleConfig(ctx context.Context, dsp *dspa.DataSciencePipelinesApplication, params *DSPAParams) (string, error) {
+	// TODO(gfrasca): do this more systematically and/or extendably
+	// enableInstructLabPipeline, err := r.IsPipelineEnabledByPlatform("instructlab")
+	// if err != nil {
+	// 	return "", err
+	// }
+	// enableIrisPipeline, err := r.IsPipelineEnabledByPlatform("iris")
+	// if err != nil {
+	// 	return "", err
+	// }
+
+	// Check if InstructLab Pipeline enabled in this DSPA
+	enableInstructLabPipeline := false
+	if dsp.Spec.APIServer.ManagedPipelines != nil && dsp.Spec.APIServer.ManagedPipelines.InstructLab != nil {
+		settingInDSPA := dsp.Spec.APIServer.ManagedPipelines.InstructLab.State
+		if settingInDSPA != "" {
+			enableInstructLabPipeline = strings.EqualFold(settingInDSPA, "Managed")
+		}
+	}
+
+	return r.GenerateSampleConfigJSON(enableInstructLabPipeline, dsp.Spec.APIServer.EnableSamplePipeline)
+}
+
+func (r *DSPAReconciler) IsPipelineEnabledByPlatform(pipelineName string) (bool, error) {
+	var platformManagedPipelines map[string]map[string]string
+	platformPipelinesJSON := config.GetStringConfigWithDefault("ManagedPipelines", config.DefaultManagedPipelines)
+
+	err := json.Unmarshal([]byte(platformPipelinesJSON), &platformManagedPipelines)
+	if err != nil {
+		return false, err
+	}
+
+	for name, val := range platformManagedPipelines {
+		if strings.EqualFold(name, pipelineName) {
+			return strings.EqualFold(val["state"], "Managed"), nil
+		}
+	}
+	return false, nil
+}
+
+func (r *DSPAReconciler) GenerateSampleConfigJSON(enableInstructLabPipeline, enableIrisPipeline bool) (string, error) {
+
+	// Now generate a sample config
+	var pipelineConfig = make([]map[string]string, 0)
+	if enableInstructLabPipeline {
+		item, err := r.GenerateSamplePipelineMetadataBlock("instructlab")
+		if err != nil {
+			return "", err
+		}
+		pipelineConfig = append(pipelineConfig, item)
+	}
+	if enableIrisPipeline {
+		item, err := r.GenerateSamplePipelineMetadataBlock("iris")
```

Comment:
The casing don't match what I see in the config. Is that expected?

Reply:
at the end of the day this is passed to viper which is case insensitive, so this could be any of `instructlab`, `INSTRUCTLAB`, `InstructLab`, etc and still return the same data from the DSPO Config